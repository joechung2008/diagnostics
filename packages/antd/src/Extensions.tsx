import { Menu, type MenuProps } from "antd";
import { byKey, isExtensionInfo, toNavLink } from "./utils";

const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const links = Object.values(extensions)
    .filter(isExtensionInfo)
    .map(toNavLink)
    .sort(byKey);

  const items = links.map((link) => ({
    key: link.key,
    label: link.name,
  }));

  const handleMenuClick: MenuProps["onClick"] = (info) => {
    const link = links.find(({ key }) => key === info.key);
    if (link) {
      onLinkClick?.(info.domEvent, link);
    }
  };

  return (
    <div className="extension-root">
      <Menu
        aria-label="Extensions"
        items={items}
        mode="vertical"
        onClick={handleMenuClick}
      />
    </div>
  );
};

export default Extensions;
