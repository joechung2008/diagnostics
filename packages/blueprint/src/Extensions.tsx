import { Tab, Tabs } from "@blueprintjs/core";
import { useState } from "react";
import { byKey, isExtensionInfo, toNavLink } from "./utils";

const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const links = Object.values(extensions)
    .filter(isExtensionInfo)
    .map(toNavLink)
    .sort(byKey);

  const [selectedTabId, setSelectedTabId] = useState<string>();

  return (
    <div className="extension-root">
      <Tabs
        vertical
        selectedTabId={selectedTabId}
        onChange={(tabId: string) => {
          setSelectedTabId(tabId);
          const link = links.find((l) => l.key === tabId);
          if (link) {
            onLinkClick(link);
          }
        }}
      >
        {links.map((link) => (
          <Tab
            key={link.key}
            aria-controls="extension"
            id={link.key}
            title={link.name}
          />
        ))}
      </Tabs>
    </div>
  );
};

export default Extensions;
