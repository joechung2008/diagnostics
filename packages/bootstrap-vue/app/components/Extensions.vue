<template>
  <nav class="extensions" aria-label="Extensions">
    <div class="d-flex flex-column gap-3">
      <BLink
        v-for="link in links"
        :key="link.key"
        href="#"
        class="text-start"
        @click="handleClick(link)"
      >
        {{ link.name }}
      </BLink>
    </div>
  </nav>
</template>

<script setup lang="ts">
import type { ExtensionsProps, KeyedNavLink } from "@/types/ExtensionsProps";
import { byKey, isExtensionInfo, toNavLink } from "@/utils/extensions";

const props = defineProps<ExtensionsProps>();

const links = computed(() =>
  Object.values(props.extensions ?? {})
    .filter(isExtensionInfo)
    .map(toNavLink)
    .sort(byKey)
);

function handleClick(link: KeyedNavLink) {
  props.onLinkClick?.(link);
}
</script>

<style scoped>
.extensions {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0.5rem;
}
</style>
