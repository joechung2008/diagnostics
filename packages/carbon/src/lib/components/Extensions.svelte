<script lang="ts">
	import { byKey, isExtensionInfo, toNavLink } from '$lib';
	import { SideNavLink } from 'carbon-components-svelte';

	let { extensions, onLinkClick }: ExtensionsProps = $props();

	let links = $derived(
		Object.values(extensions).filter(isExtensionInfo).map(toNavLink).sort(byKey)
	);
</script>

{#each links as link (link.key)}
	<SideNavLink text={link.name} on:click={() => onLinkClick(link)} />
{/each}
