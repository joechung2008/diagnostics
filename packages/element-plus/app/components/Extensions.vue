<template>
    <nav class="extensions" aria-label="Extensions">
        <el-button v-for="link in links" :key="link.key" class="extension" @click="handleLinkClick(link)">
            {{ link.name }}
        </el-button>
    </nav>
</template>

<script setup lang="ts">
import type { ExtensionsProps, KeyedNavLink } from "@/types"
import { byKey, isExtensionInfo, toNavLink } from "../utils/extensions"

const props = defineProps<ExtensionsProps>()

const links = computed(() =>
    Object.values(props.extensions)
        .filter(isExtensionInfo)
        .map(toNavLink)
        .sort(byKey)
)

function handleLinkClick(link: KeyedNavLink) {
    props.onLinkClick(link)
}
</script>

<style scoped>
.extensions {
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    height: 100%;
    overflow-y: auto;
    padding-right: 0.5rem;
}

/* Override Element Plus default margin */
.extension {
    justify-content: start;
    margin: 0;
}
</style>
