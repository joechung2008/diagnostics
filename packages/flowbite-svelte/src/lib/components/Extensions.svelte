<script lang="ts">
  import { byKey, isExtensionInfo, toNavLink } from "$lib/utils";
  import { Sidebar, SidebarGroup, SidebarItem } from "flowbite-svelte";

  let { extensions, onLinkClick }: ExtensionsProps = $props();

  const links = $derived.by(() =>
    Object.values(extensions).filter(isExtensionInfo).map(toNavLink).sort(byKey)
  );
</script>

<div class="flex flex-none flex-col gap-0.5 overflow-y-auto p-1">
  <Sidebar alwaysOpen aria-label="Extensions" position="static" class="!z-0 w-auto">
    <SidebarGroup>
      {#each links as link (link.key)}
        <SidebarItem
          label={link.name}
          onclick={() => onLinkClick?.(link)}
          class="cursor-pointer"
          tabindex={0}
        />
      {/each}
    </SidebarGroup>
  </Sidebar>
</div>
