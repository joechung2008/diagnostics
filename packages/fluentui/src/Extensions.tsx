import { Button, makeStyles, mergeClasses } from "@fluentui/react-components";
import { useMemo } from "react";
import { byKey, isExtensionInfo, toNavLink } from "./utils";

const useStyles = makeStyles({
  navButton: {
    justifyContent: "flex-start",
    textAlign: "left",
    width: "100%",
    minHeight: "max-content",
    boxSizing: "border-box",
  },
});
const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const styles = useStyles();
  const links = useMemo(
    () =>
      Object.values(extensions)
        .filter(isExtensionInfo)
        .map(toNavLink)
        .sort(byKey),
    [extensions]
  );

  return (
    <nav className="extension-root" aria-label="Extensions">
      {links.map((link) => (
        <Button
          key={link.key}
          className={mergeClasses("extension-nav-button", styles.navButton)}
          appearance="subtle"
          onClick={(e) => onLinkClick?.(e, link)}
        >
          {link.name}
        </Button>
      ))}
    </nav>
  );
};

export default Extensions;
