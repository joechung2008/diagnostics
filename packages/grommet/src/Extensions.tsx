import { byKey, isExtensionInfo, toNavLink } from "./utils";
import { Nav, Button } from "grommet";

const Extensions: React.FC<ExtensionsProps> = ({ extensions, onLinkClick }) => {
  const links = Object.values(extensions)
    .filter(isExtensionInfo)
    .map(toNavLink)
    .sort(byKey);

  return (
    <Nav
      aria-label="Extensions"
      className="extension-root overflow-x-hidden"
      direction="column"
      gap="small"
      pad="small"
    >
      {links.map((link) => (
        <Button
          key={link.key}
          label={link.name}
          plain
          onClick={(e) => onLinkClick?.(e, link)}
        />
      ))}
    </Nav>
  );
};

export default Extensions;
