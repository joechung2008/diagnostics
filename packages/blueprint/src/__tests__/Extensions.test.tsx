import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Extensions from "../Extensions";

describe("Extensions", () => {
  it("matches snapshot", () => {
    const mockExtensions = { ext1: { extensionName: "SampleExtension" } };
    const mockOnLinkClick = () => {};
    const { asFragment } = render(
      <Extensions extensions={mockExtensions} onLinkClick={mockOnLinkClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
