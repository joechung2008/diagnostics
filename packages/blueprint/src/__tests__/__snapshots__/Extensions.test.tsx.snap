// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extensions > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="extension-root"
  >
    <div
      class="bp6-tabs bp6-vertical"
    >
      <div
        class="bp6-tab-list"
        role="tablist"
      >
        <div
          class="bp6-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px; transition: none;"
        >
          <div
            class="bp6-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp6-tab-panel_undefined_SampleExtension"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp6-tab"
          data-tab-id="SampleExtension"
          id="bp6-tab-title_undefined_SampleExtension"
          role="tab"
          tabindex="0"
        >
          SampleExtension
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
